### PR TITLE
Amend NameAndTypeResolution expected messages

### PR DESCRIPTION
**Pull request by izgzhen**
Fixes https://github.com/ethereum/solidity/issues/2613.

### CODE REVIEW

**Comment by axic** [comment]
Thanks! A few tests fail, see here for a list: https://travis-ci.org/ethereum/solidity/jobs/256799732

**Comment by axic** [comment]
Oh I see the probable reason. @izgzhen these aren't messages displayed to the user during testing, but the messages the compiler emits when compiling those code snippets. Our tests need to ensure we check that the right message was emitted by the compiler.

**Comment by izgzhen** [comment]
Right ... so I think another half of the work is still missing right?

**Comment by izgzhen** [comment]
If so, although not quite the "up-for-grab" level, I'll give a try to some of them at least 😆

**Comment by chriseth** [comment]
@izgzhen so the idea is that you copy-paste the source code into a file, run `solc` on the file, check that there is exactly one error message and just copy the error message into the test file. If you want to verify, run `soltest -- --no-ipc`.

**Comment by izgzhen** [comment]
Fixed, and for some cases, I found that no matter what I filled in, it is always okay (I filled in `"x"`)

**Comment by axic** [comment]
> for some cases, I found that no matter what I filled in, it is always okay (I filled in "x")

Are you sure the message doesn't contain a single letter `x` in those cases?

It tries a text submatch.

**Comment by izgzhen** [comment]
> Are you sure the message doesn't contain a single letter x in those cases?
> It tries a text submatch.

Well, it is interesting .. I'll try it again. Thanks!

**Comment by izgzhen** [comment]
@axic Fixed, sorry that I was a bit busy these days thus I didn't carefully go through the existing code before diving into it. Thanks a lot for your patience

**Comment by axic** [comment]
Thanks! Is this still partially solving #2613 or does it change every message?

**Comment by izgzhen** [comment]
> Thanks! Is this still partially solving #2613 or does it change every message?

Every originally `""` msg that I can find.

**Comment by axic** [comment]
Looks good to me.

**Comment by axic** [comment]
@chriseth please double check, but would be nice to merge soon

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 1]
```diff
@@ -1166,7 +1166,7 @@ BOOST_AUTO_TEST_CASE(function_overrides_modifier)
 	)";
 	// Error: Identifier already declared.
 	// Error: Override changes function to modifier.
-	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "");
+	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "Identifier already declared.");
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp:1131]: Actually the real test case here is "Override changes function to modifier.".

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 2]
```diff
@@ -1155,7 +1155,7 @@ BOOST_AUTO_TEST_CASE(modifier_overrides_function)
 	)";
 	// Error: Identifier already declared.
 	// Error: Override changes modifier to function.
-	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "Identifier already declared");
+	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "Identifier already declared.");
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp:1120]: Actually the real test case here is "Override changes modifier to function."

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 3]
```diff
@@ -1404,7 +1404,7 @@ BOOST_AUTO_TEST_CASE(fallback_function_twice)
 			function() { x = 3; }
 		}
 	)";
-	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "");
+	CHECK_ERROR_ALLOW_MULTI(text, DeclarationError, "Function with same name and arguments defined twice.");
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp:1369]: The real test case here is "Only one fallback function is allowed."

**Comment by axic** [comment]
@izgzhen sorry it took so long, but could you change the above three things? I'd like to get this merged asap.

**Comment by izgzhen** [comment]
> @izgzhen sorry it took so long, but could you change the above three things? I'd like to get this merged asap.

It is okay :) I've updated the code, hoping it works.

**Comment by axic** [comment]
Thanks!

Seems like we need to improve our testing infrastructure to handle those properly, because it always takes the first error:

```
Expected message "Override changes function to modifier." but found "Identifier already declared.".
/home/travis/build/ethereum/solidity/test/libsolidity/SolidityNameAndTypeResolution.cpp(1120): error in "modifier_overrides_function": check searchErrorMessage(err, ("Override changes function to modifier.")) failed
Expected message "Override changes function to modifier." but found "Identifier already declared.".
/home/travis/build/ethereum/solidity/test/libsolidity/SolidityNameAndTypeResolution.cpp(1131): error in "function_overrides_modifier": check searchErrorMessage(err, ("Override changes function to modifier.")) failed
Expected message "Only one fallback function is allowed." but found "Function with same name and arguments defined twice.".
/home/travis/build/ethereum/solidity/test/libsolidity/SolidityNameAndTypeResolution.cpp(1369): error in "fallback_function_twice": check searchErrorMessage(err, ("Only one fallback function is allowed.")) failed
```

Can you just revert those 3 to what you had and I'll address it in a separate PR?

**Comment by axic** [comment]
@izgzhen realised I can push to your branch and did the changes myself

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
@izgzhen thanks!

**Pull request merged**